log/logger_file.go

```diff
@@ -0,0 +1,39 @@
+package log
+
+import (
+	"github.com/op/go-logging"
+	"os"
+)
+
+// FileLogger outputs logs to a given file.
+type FileLogger struct {
+	basicLogger
+}
+
+func NewFileLogger(module, logFilePath, logLevel, format string) (*FileLogger, error) {
+	basicLogger := newBasicLogger(module)
+
+	levelInt, err := logging.LogLevel(logLevel)
+	if err != nil {
+		return nil, err
+	}
+	logFile, err := os.OpenFile(logFilePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
+	if err != nil {
+		return nil, err
+	}
+	backend, err := basicLogger.setupFormattedBackend(logFile, format)
+	if err != nil {
+		return nil, err
+	}
+
+	leveledBackend := logging.SetBackend(backend)
+	leveledBackend.SetLevel(levelInt, module)
+	logger := &FileLogger{basicLogger}
+	logger.SetBackend(leveledBackend)
+
+	return logger, nil
+}
+
+func GetDefaultFileLogger(module, logfilePath string) (*FileLogger, error) {
```

Comment:
The `Default` in function name probably refers to default level and format being used? Even so, it's quite specialized already by `module` and `logFileLocation` (<-- camelCase 😈 ) so it might be better to shorten the name into:

```go
GetFileLogger
```

WDYT?

Reply:
EDIT: On a second thought, do we really need this function at all? I think whenever we use it, we then set logging level manually in the very next line with `SetLogLevel()`. I may be missing something, but wouldn't it be better to use the `NewFileLogger` in first place?

Reply:
Indeed. I went with your second comment and removed wrapper functions altogether.